BAM
===

**Format version:** {{spec_value("bam", "file.version")}}

[Binary Alignment Map](https://en.wikipedia.org/wiki/Binary_Alignment_Map) (BAM) is the comprehensive raw data of genome sequencing; it consists of the lossless, compressed binary representation of the Sequence Alignment Map-files.

BAM files generated by MinKNOW are compliant with the [SAM Specification](https://samtools.github.io/hts-specs/SAMv1.pdf), and can be manipulated using [Samtools](http://www.htslib.org/).

Paths
-----

The following path patterns are used to place the data on disk:

File         | Path pattern
------------ | ------------
**BAM file** | ``{{spec_value("experiment_layout", "protocol_spec.contents.bam.bam_path")}}``
**BAI file** | ``{{spec_value("experiment_layout", "protocol_spec.contents.bam.bai_path")}}``

See the [Patterns](../minknow/patterns.md) documentation for more information on file patterns.

Read batching
-------------

The following batching options are used by default:


Option       | Value
------------ | -----
**Duration** | ``{{spec_value("experiment_layout", "protocol_spec.contents.bam.batching.duration")}}``

For more information on batching see [Batching](../minknow/batching.md).

Fields
------

### Read groups

::: bam_spec_docs header_records RG

### Program groups

::: bam_spec_docs header_records PG

### Read tags

::: bam_spec_docs read_tags
